test/unit/data/events/scheme/get-events-by-scheme-id.test.js

```diff
@@ -0,0 +1,57 @@
+jest.mock('../../../../../app/data/events/scheme-id/get-submitted-events')
+const { getSubmittedEvents: mockGetSubmittedEvents } = require('../../../../../app/data/events/scheme-id/get-submitted-events')
+
+jest.mock('../../../../../app/data/events/scheme-id/group-events-by-scheme')
+const { groupEventsByScheme: mockGroupEventsByScheme } = require('../../../../../app/data/events/scheme-id/group-events-by-scheme')
+
+jest.mock('../../../../../app/data/events/scheme-id/order-grouped-events-by-scheme')
+const { orderGroupedEventsByScheme: mockOrderGroupedEventsByScheme } = require('../../../../../app/data/events/scheme-id/order-grouped-events-by-scheme')
+
+jest.mock('../../../../../app/data/events/scheme-id/get-total-scheme-values')
+const { getTotalSchemeValues: mockGetTotalSchemeValues } = require('../../../../../app/data/events/scheme-id/get-total-scheme-values')
+
+jest.mock('../../../../../app/data/events/scheme-id/sanitise-scheme-data')
+const { sanitiseSchemeData: mockSanitiseSchemeData } = require('../../../../../app/data/events/scheme-id/sanitise-scheme-data')
+
+const { SCHEME_ID: SCHEME_ID_VALUE } = require('../../../../mocks/values/scheme-id')
+const { SCHEME_ID: SCHEME_ID_CATEGORY } = require('../../../../../app/constants/categories')
+const submitted = require('../../../../mocks/events/submitted')
+const groupedEvent = require('../../../../mocks/events/grouped-event')
+const totalSchemeValues = require('../../../../mocks/events/total-scheme-values')
+
+const { getEventsBySchemeId } = require('../../../../../app/data/events/scheme-id/get-events-by-scheme-id')
+
+describe('get events by frn', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
```

Comment:
removing this `jest.clearAllMocks()` allows all tests to pass. Currently they all fail. Where is the correct place to rest the mocks or do i not need to in this scenario?

Reply:
needed an `await` on each test

Reply:
So with the clearAllMocks the tests fail?

Reply:
no, they pass now i had missed off `await` when calling the function `getEventsBySchemeId` in the individual tests. I make this mistake roughly once a month 😅

Reply:
You're fine!